### PR TITLE
Passport

### PR DESCRIPTION
**Pull request by cartige**

### CODE REVIEW

File: frontend/src/components/Registration/Registration.jsx [hunk 1]
```diff
@@ -17,16 +16,25 @@ export default function Registration() {
...
-    password: "",
-    premium: 0,
-  });
+  // const [registrationForm, setRegistrationForm] = useState({
+  //   username: "",
+  //   lastname: "",
+  //   email: "",
+  //   password: "",
+  //   premium: 0,
+  // });
```
Comment (CharlyJuteau) [inline frontend/src/components/Registration/Registration.jsx]: dead code

File: frontend/src/components/Registration/Registration.jsx [hunk 2]
```diff
@@ -40,17 +48,7 @@ export default function Registration() {
...
-    axios
-      .post("http://localhost:5000/users", registrationForm)
-      .then(({ data }) => {
-        setCurrentUser(data);
-      })
-      .catch((err) => {
-        console.error(err);
-      });
-  };
+  console.log(currentUser);
```
Comment (CharlyJuteau) [inline frontend/src/components/Registration/Registration.jsx]: console.log

File: frontend/src/components/ViewProfil/ViewProfil.jsx [hunk 3]
```diff
@@ -5,28 +5,14 @@ import "./viewProfil.scss";
...
-    axios
-      .put(`http://localhost:5000/users/${currentUser.id}`, userProfil)
-      .then(({ data }) => {
-        setCurrentUser(data);
-      })
-      .catch((err) => {
-        console.error("Error editing the user", err);
-      });
-  };
+  console.log(userProfil);
```
Comment (CharlyJuteau) [inline frontend/src/components/ViewProfil/ViewProfil.jsx]: console.log

File: frontend/src/contexts/UserContext.jsx [hunk 4]
```diff
@@ -1,47 +1,143 @@
...
+    password: "",
+    premium: 0,
+  });
+
   const [loginForm, setLoginForm] = useState({ username: "", password: "" });
 
-  const hLogin = (evt) => {
+  const [userProfil, setUserProfil] = useState(currentUser);
+
+  // const { token } = currentUser;
```
Comment (CharlyJuteau) [inline frontend/src/contexts/UserContext.jsx]: dead code

File: frontend/src/contexts/UserContext.jsx [hunk 5]
```diff
@@ -1,47 +1,143 @@
...
+    //   .then(({ data }) => {
+    //     console.log(data);
+    //     const { token, user } = data;
+    //     api.defaults.headers.authorization = `Bearer ${token}`;
+    //     setIsAuthenticated(true);
+    //     setCurrentUser(user);
+    //   })
+    //   .catch((err) => {
+    //     console.error("Error while logging in", err);
+    //   });
```
Comment (CharlyJuteau) [inline frontend/src/contexts/UserContext.jsx]: dead code

File: frontend/src/contexts/UserContext.jsx [hunk 6]
```diff
@@ -1,47 +1,143 @@
...
+    //   .then(({ data }) => {
+    //     const { token, user } = data;
+
+    //     api.defaults.headers.authorization = `Bearer ${token}`;
+    //     setIsAuthenticated(true);
+    //     setCurrentUser(user);
+    //   })
+    //   .catch((err) => {
+    //     console.error(err);
+    //   });
```
Comment (CharlyJuteau) [inline frontend/src/contexts/UserContext.jsx]: dead code

File: frontend/src/contexts/UserContext.jsx [hunk 7]
```diff
@@ -1,47 +1,143 @@
...
+
+  const hEditFormSubmit = (evt) => {
     evt.preventDefault();
-    axios
-      .post("http://localhost:5000/login", loginForm)
+    api
+      .put(`/users/${currentUser.id}`, userProfil)
       .then(({ data }) => {
-        setCurrentUser(data.user);
+        // const { token } = currentUser;
```
Comment (CharlyJuteau) [inline frontend/src/contexts/UserContext.jsx]: dead code

File: frontend/src/contexts/UserContext.jsx [hunk 8]
```diff
@@ -1,47 +1,143 @@
...
       })
       .catch((err) => {
-        console.error("Error while logging in", err);
+        console.error("Error editing the user", err);
       });
   };
 
+  useEffect(() => {
+    setUserProfil(currentUser);
+    // console.log(currentUser);
```
Comment (CharlyJuteau) [inline frontend/src/contexts/UserContext.jsx]: dead code

File: frontend/src/contexts/UserContext.jsx [hunk 9]
```diff
@@ -1,47 +1,143 @@
...
+    console.log("user profil : ", userProfil);
```
Comment (CharlyJuteau) [inline frontend/src/contexts/UserContext.jsx]: console.log

File: frontend/src/contexts/UserContext.jsx [hunk 10]
```diff
@@ -1,47 +1,143 @@
...
+      registrationForm,
+      userProfil,
+      isAuthenticated,
+      setUserProfil,
+      setRegistrationForm,
+      setLoginForm,
+      setCurrentUser,
+    ]
   );
+  // console.log(currentUser);
```
Comment (CharlyJuteau) [inline frontend/src/contexts/UserContext.jsx]: dead code

File: frontend/src/pages/Favoris/Favoris.jsx [hunk 11]
```diff
@@ -13,20 +18,10 @@ import NavBar from "../../components/NavBar/NavBar";
...
-  }, []);
+  const { currentUser, isAuthenticated } = useContext(UserContext);
 
-  useEffect(() => {
-    axios.get("http://localhost:5000/favoris/1").then(({ data }) => {
-      setVideos(data);
-    });
-  }, []);
+  // console.log(currentUser);
+  // console.log(isAuthenticated);
```
Comment (CharlyJuteau) [inline frontend/src/pages/Favoris/Favoris.jsx]: dead code

**Review by CharlyJuteau — DISMISSED** [review]

File: backend/src/app.js [hunk 12]
```diff
@@ -16,6 +20,17 @@ app.use(
 
 app.use(express.json());
 
+app.use(
+  session({
+    secret: "keyboard cat",
```
Comment (LoicBrassart) [inline backend/src/app.js]: Set your secret variables in your .env :pray:

File: frontend/src/pages/Favoris/Favoris.jsx [hunk 13]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (LoicBrassart) [inline frontend/src/pages/Favoris/Favoris.jsx]: Why add an empty line here ?

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Some Evils!

**Review by CharlyJuteau — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
Better!

**Pull request merged**